internal/provider/config_resource.go

```diff
@@ -214,6 +214,28 @@ type ConditionalHeaderModel struct {
 	NoMatchValue   types.String `tfsdk:"no_match_value"`
 }
 
+func (c *ConditionalHeaderModel) transformToDto() client.ConditionalHeaderDto {
+	var condHeadPostBody = client.ConditionalHeaderDto{
+		MatchingHeader: c.MatchingHeader.ValueString(),
+		Pattern:        c.Pattern.ValueString(),
+		NewHeader:      c.NewHeader.ValueString(),
+		MatchValue:     c.MatchValue.ValueString(),
+		NoMatchValue:   c.NoMatchValue.ValueString(),
+	}
+	return condHeadPostBody
+}
+
+func transformCondHeaderToResourceModel(c *client.ConditionalHeaderDto) ConditionalHeaderModel {
+	var condHeaderResourceModel = ConditionalHeaderModel{
+		MatchingHeader: types.StringValue(c.MatchingHeader),
```

Comment:
ConditionalHeaderDto and ConditionalHeaderModel have the same fields, is it possible just to use one of these?

Reply:
I believe they need to be two separate models due to the way the go code understands them internally, the annotations on the end of each of the types, one interchanges from json to a go struct and one is the same but between go struct and a tf resource.